packages/core/src/engines/query/JSONQueryEngine.ts

```diff
@@ -0,0 +1,178 @@
+};
+
+type OutputProcessor = typeof defaultOutputProcessor;
+
+export class JSONQueryEngine implements BaseQueryEngine {
+  jsonValue: JSONSchemaType;
+  jsonSchema: JSONSchemaType;
+  serviceContext: ServiceContext;
+  outputProcessor: OutputProcessor;
+  outputKwargs: Record<string, unknown>;
+  verbose: boolean;
+  jsonPathPrompt: JSONPathPrompt;
+  synthesizeResponse: boolean;
+  responseSynthesisPrompt: ResponseSynthesisPrompt;
+
+  constructor(init: {
+    jsonValue: JSONSchemaType;
+    jsonSchema: JSONSchemaType;
+    serviceContext?: ServiceContext;
+    jsonPathPrompt?: JSONPathPrompt;
+    outputProcessor?: OutputProcessor;
+    outputKwargs?: Record<string, unknown>;
+    synthesizeResponse?: boolean;
+    responseSynthesisPrompt?: ResponseSynthesisPrompt;
+    verbose?: boolean;
+  }) {
+    this.jsonValue = init.jsonValue;
+    this.jsonSchema = init.jsonSchema;
+    this.serviceContext = init.serviceContext ?? serviceContextFromDefaults({});
+    this.jsonPathPrompt = init.jsonPathPrompt ?? defaultJsonPathPrompt;
+    this.outputProcessor = init.outputProcessor ?? defaultOutputProcessor;
+    this.outputKwargs = init.outputKwargs ?? {};
+    this.verbose = init.verbose ?? false;
+    this.synthesizeResponse = init.synthesizeResponse ?? true;
+    this.responseSynthesisPrompt =
+      init.responseSynthesisPrompt ?? defaultResponseSynthesizePrompt;
+  }
+
+  getPrompts(): Record<string, unknown> {
+    return {
+      jsonPathPrompt: this.jsonPathPrompt,
+      responseSynthesisPrompt: this.responseSynthesisPrompt,
+    };
+  }
+
+  updatePrompts(prompts: {
+    jsonPathPrompt?: JSONPathPrompt;
+    responseSynthesisPrompt?: ResponseSynthesisPrompt;
+  }): void {
+    if (prompts.jsonPathPrompt) {
+      this.jsonPathPrompt = prompts.jsonPathPrompt;
+    }
+    if (prompts.responseSynthesisPrompt) {
+      this.responseSynthesisPrompt = prompts.responseSynthesisPrompt;
+    }
+  }
+
+  getPromptModules(): Record<string, unknown> {
+    return {};
+  }
+
+  getSchemaContext(): string {
+    return JSON.stringify(this.jsonSchema);
+  }
+
+  query(params: QueryEngineParamsStreaming): Promise<AsyncIterable<Response>>;
+  query(params: QueryEngineParamsNonStreaming): Promise<Response>;
+  async query(
+    params: QueryEngineParamsStreaming | QueryEngineParamsNonStreaming,
+  ): Promise<Response | AsyncIterable<Response>> {
+    const { query, stream } = params;
+
+    if (stream) {
+      throw new Error("Streaming is not supported");
+    }
+
+    const schema = this.getSchemaContext();
+
+    const jsonPathResponseStr = await this.serviceContext.llm.complete({
+      prompt: this.jsonPathPrompt({ query, schema }),
+    });
+
+    if (this.verbose) {
+      console.log(
+        `> JSONPath Instructions:\n\`\`\`\n${jsonPathResponseStr}\n\`\`\`\n`,
+      );
+    }
+
+    const jsonPathOutput = await this.outputProcessor({
+      llmOutput: jsonPathResponseStr.text,
+      jsonValue: this.jsonValue,
+      ...this.outputKwargs,
```

Comment:
no kwargs in TS and type `OutputProcessor` does just support `llmOutput` and `jsonValue` as params

I would leave it out